mongo/integration/crud_prose_test.go

```diff
@@ -113,6 +114,242 @@ func TestWriteErrorsWithLabels(t *testing.T) {
 
 }
 
+func TestWriteErrorsDetails(t *testing.T) {
+	clientOpts := options.Client().
+		SetRetryWrites(false).
+		SetWriteConcern(mtest.MajorityWc).
+		SetReadConcern(mtest.MajorityRc)
+	mtOpts := mtest.NewOptions().
+		ClientOptions(clientOpts).
+		MinServerVersion("5.0").
+		Topologies(mtest.ReplicaSet, mtest.Single).
+		CreateClient(false)
+
+	mt := mtest.New(t, mtOpts)
+	defer mt.Close()
+
+	mt.Run("JSON Schema validation", func(mt *mtest.T) {
```

Comment:
The tests in here feel quite repetitive. The only difference seems to be whether the operation returns `WriteException` or `BulkWriteException`. Could we transform these into table tests? I'm thinking something like:

```
name string
operationFn func(*mongo.Collection) error // callback to run insert/update/etc 
expectBulkError // if true, test will expect a BulkWriteException. Otherwise, it will expect WriteException
erroringOperationName // name of the operation that should have returned the error with an "errInfo" field
```

Reply:
Sounds good, will update the tests to use test cases instead of separate functions.